src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2673,14 +2685,18 @@ private static String buildCanonicalHeaders(Request request,
         return Joiner.on("\n").join(headersWithValues);
     }
 
-    private static String buildCanonicalQueryString(Request request)
+    private static String buildCanonicalQueryString(HttpServletRequest request)
             throws UnsupportedEncodingException {
         // The parameters are required to be sorted
         List<String> parameters = Collections.list(request.getParameterNames());
         Collections.sort(parameters);
         List<String> queryParameters = new ArrayList<>();
-        String charsetName = Objects.firstNonNull(request.getQueryEncoding(),
+
+        // TODO  Remove this as it is not used?
```

Comment:
Agree that the method does not use `charsetName` any more; please remove it in this commit.